_posts/blog/2016-04-15-subscribe_on.md

```diff
@@ -0,0 +1,83 @@
+---
+layout: post
+title: "RxJava: subscribeOn vs observeOn"
+modified:
+categories: blog
+author: tomek_polanski
+excerpt:
+tags: [RxJava]
+image:
+date: 2016-04-14T13:00:55-01:00
+---
+
+
+## RxJava: subscribeOn vs observeOn
+
+One of the strongest aspects of RxJava is the simple way to schedule work on a desired thread using either ``subscribeOn`` or ``observeOn``. Understanding how each one works is crucial.  
+
+
+### observeOn:
+This method simply changes the thread of all operators further **downstream** (in the calls that come after). Let’s assume code is run from a UI thread: 
+
+<picture>
+	<img src="/images/ObserveOn.gif" alt="image">
+</picture>
+
+One of the most frequent misconceptions is that ``observeOn`` also acts upstream, but really it acts **only** downstream - things that happen after the ``observeOn`` call - unlike ``subscribeOn``.  
+
+### subscribeOn:
+This only influences the thread that is used when the observable is subscribed to and it will stay on it until changed.  
```

Comment:
> it will stay on it until changed. 

It should be clarified that this is in context of the downstream from the source Observable. Use the downstream here.